h2c/server.go

```diff
@@ -19,6 +24,9 @@ type Server struct {
 	// use the Upgrade header in this case. This is not to spec, but seems to
 	// work OK.
 	ALBSupport bool
+
+	connections map[net.Conn]struct{}
+	mutex       sync.Mutex
```

Comment:
this a pretty generic name, consider `connectionsMu` to make it clear what it's for